src/App.tsx

```diff
@@ -67,6 +64,17 @@ function App() {
     setDisplayNewCard(false);
   };
 
+  const updateCard = (id: number, cardSide: string, input: string) => {
+    const updateCard = cardsData.map((card) => {
+      if (card.id === id) {
+        return { ...card, cardSide: input }
```

Comment:
**issue:** Logic is correct but you have bug here, as you pass `cardSide` incorrectly. You should close it in `[]` to have dynamic key.

```js
 const updateCard = (id: number, cardSide: string, input: string) => {
    const updateCard = cardsData.map((card) => {
      if (card.id === id) {
        return { ...card, [cardSide]: input }
      }
      return card
    });

    setCardsData(updateCard);
    }
```